File: Data Structures & Algorithms/maximum-subarray/submission-2.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        sum_ = 0
        max_sum = nums[0]

        for i in range(len(nums)):
            sum_ += nums[i]
            max_sum = max(max_sum, sum_)
            if sum_ < 0:
                sum_ = 0
            
        return max_sum

File: Data Structures & Algorithms/jump-game/submission-2.java
class Solution {
    public boolean canJump(int[] nums) {
        int max_i = 0;
        for (int i=0; i<nums.length; i++) {
            if (max_i < i) return false;

            int curr_i = nums[i] + i;
            max_i = Math.max(max_i, curr_i);
        }
        return max_i >= nums.length-1;
    }
}
